src/append_only_data.rs

```diff
@@ -1398,11 +1398,11 @@ mod tests {
         for data in data_vec {
             assert_eq!(
                 data.entry(Index::FromStart(0)),
-                Some(&Entry::new(b"key0".to_vec(), b"value0".to_vec()))
+                Some(&entries.clone()[0])
```

Comment:
Won't this clone the entire vec? I think `entries[0].clone()` is what you want, but I haven't tested it.

Of course, this is just a test so the performance here isn't important.

Reply:
If the entries inside the `Vec` are a `Clone` type, then they will be cloned (even if they are a reference), otherwise if they are a reference but not a `Clone` type the references themselves will be cloned (shallow clone), otherwise you get a compiler error